sdk/python/kfp/_client.py

```diff
@@ -99,16 +99,29 @@ def list_experiments(self, page_token='', page_size=10, sort_by=''):
         page_token=page_token, page_size=page_size, sort_by=sort_by)
     return response
 
-  def get_experiment(self, experiment_id):
+  def get_experiment(self, experiment_id=None, experiment_name=None):
     """Get details of an experiment
+    Either experiment_id or experiment_name is required
     Args:
-      id of the experiment.
+      experiment_id: id of the experiment. (Optional)
+      experiment_name: name of the experiment. (Optional)
     Returns:
       A response object including details of a experiment.
     Throws:
-      Exception if experiment is not found.        
+      Exception if experiment is not found or None of the arguments is provided
     """
-    return self._experiment_api.get_experiment(id=experiment_id)
+    if experiment_id is None and experiment_name is None:
+      raise ValueError('Either experiment_id or experiment_name is required')
+    if experiment_id is not None:
+      return self._experiment_api.get_experiment(id=experiment_id)
+    next_page_token = ''
```

Comment:
What do you think about having a separate `get_experiment_by_name` function? It would be a bit shorter and the code will be simpler?

    get_experiment_by_name('foo')
vs

    get_experiment(experiment_name='foo')

Reply:
Bradley and I discussed this yesterday. It seems more natural and simple to have get_experiment that takes either name or id. Or else, it would rather be two functions named get_experiment_by_id and get_experiment_by_name, which would break the backward compatibility because we still want to support the get_experiment function.